### PR TITLE
Update DCR Dependencies (workspaces prep)

### PR DESCRIPTION
**Pull request by OllysCoding**
## What does this change?

Update all DCR dependencies & dev dependencies in line with the allowances of the package.json.

Fixes related to package updates:
- Remove unused props (ESLint Error)
- Use `@ts-expect-error` for new typescript errors which couldn't be resolved

## Why?

When we pull DCR & AR into yarn workspaces in the root of the monorepo, our lock files will be lost. By upgrading our packages to the most recent minor/patch versions available, when a new lock file is created, we'll have already fixed any associated bugs.

### CODE REVIEW

**Comment by OllysCoding** [comment]
@LinusU Cheers mate!

**Review by nitro-marky — APPROVED** [review]

File: dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx [hunk 1]
```diff
@@ -86,7 +86,6 @@ const Headline: React.FC<{
 	headlineText: string;
 	standfirst: string;
 	pillar: Theme;
-	starRating?: number;
```
Comment (lucymonie) [inline dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx:89]: What's the reason that star rating has been removed as part of this work?

File: dotcom-rendering/src/web/components/ArticleHeadline.stories.tsx [hunk 2]
```diff
@@ -97,7 +97,6 @@ export const ShowcaseInterview = () => (
 							theme: Pillar.Culture,
 						}}
 						tags={[]}
-						isShowcase={true}
```
Comment (lucymonie) [inline dotcom-rendering/src/web/components/ArticleHeadline.stories.tsx:100]: Just wondering why this is getting removed?

**Review by lucymonie — COMMENTED** [review]
I just have some questions about removal of props - was this to do with the unused props error that you mentioned in your notes? They seem to be in use.

File: dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx (hunk 1 above)
Comment (OllysCoding) [reply dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx:89]: This prop was not used anywhere within the `topMetaLiveblog` component, after updating, ESLint identified it and threw an error as unused prop.

Comment (OllysCoding) [reply dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx:89]: (Despite it being passed in by a parent component, it isn't used within the component, making it redundant)

File: dotcom-rendering/src/web/components/ArticleHeadline.stories.tsx (hunk 2 above)
Comment (OllysCoding) [reply dotcom-rendering/src/web/components/ArticleHeadline.stories.tsx:100]: While the data is passed into the component, the prop isn't actually consumed anywhere within it, meaning it's essentially redundant, causing ESLint to throw an error

File: dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx (hunk 1 above)
Comment (lucymonie) [reply dotcom-rendering/src/amp/components/topMeta/TopMetaLiveblog.tsx:89]: Ah, got it.

File: dotcom-rendering/src/web/components/ArticleHeadline.stories.tsx (hunk 2 above)
Comment (lucymonie) [reply dotcom-rendering/src/web/components/ArticleHeadline.stories.tsx:100]: 👍🏻

**Review by lucymonie — APPROVED** [review]

**Review by marjisound — COMMENTED** [review]
I noticed the following packages may need another update as they're not on their maximum semver version. Could you please have a look at them?
```
eslint-plugin-react
aws-sdk
@guardian/src-text-input
@guardian/src-text-area
@guardian/src-radio
@guardian/src-link
@guardian/src-icons
@guardian/src-foundations
@guardian/src-ed-lines
@guardian/src-button
```

**Review by marjisound — APPROVED** [review]

**Comment by OllysCoding** [comment]
Thanks @marjisound, for `eslint-plugin-react` and `aws-sdk` I've checked and these two are okay. I have now updated all source libs!

**Pull request merged**